### DISCUSSION TITLE
Is it possible to render a page to an HTML string? (for printing to PDF with puppeteer)

### DISCUSSION THREAD

**Discussion by wilcoschoneveld** [vercel/next.js#19186] [Help]
I'm looking to make one of the pages available as a pdf, and I want to use puppeteer.

I need an HTML string (with styles) to inject into a puppeteer page. I found this blog post:
https://medium.com/knowsi/exporting-pdfs-with-next-js-714735f0a473

but I feel there must be a simpler way to do this? Any ideas?

Thanks!

**Comment by si4dev** [comment]
I'm also searching for a way to render a nextjs page on the server and use the output as a string. For now what I did was to to let puppeteer request the page like an external url with page.goto(url); add a security key to the url to ensure no public access.

One big issue I do experience is that sometimes the invoice page is empty. Probably the server rendering (I use graphql) was not loaded in time. Really annoying. I'm still very interested to render on the server as a string.

**Comment by martin-dimi** [comment]
Have you guys found any new solutions? cc @wilcoschoneveld

**Reply by wilcoschoneveld** [reply]
Not that I remember, and I have since moved onto other things
